hnn_core/dipole.py

```diff
@@ -137,9 +137,27 @@ def simulate_dipole(net, n_trials=1, n_jobs=1):
     dpl: list | instance of Dipole
         The dipole object or list of dipole objects if n_trials > 1
     """
-    parallel, myfunc = _parallel_func(_clone_and_simulate, n_jobs=n_jobs)
-    out = parallel(myfunc(net.params, idx) for idx in range(n_trials))
-    dpl, spiketimes, spikegids = zip(*out)
+
+    from .parallel import create_parallel_context, get_nhosts
+
+    if n_jobs > 1:
+        # check whether NEURON is using parallel nrniv processes
+        create_parallel_context(n_cores=1)
+        if get_nhosts() > 1:
+            print("Nested parallelism is not currently supported!\n" +
```

Comment:
use `ValueError`

Reply:
I don't understand. There's no exception being thrown. Note the function "get_nhosts()" is from parallel.py and is querying the state from ParallelContext.

Reply:
I see what you mean. Fixed